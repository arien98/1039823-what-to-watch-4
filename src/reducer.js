import {filmsData, promoFilm} from "./mocks/films.js";
import {ScreenMode, TabsType, extend} from "./common.js";

const ActionType = {
  // FILTER: `filter films`,
  // SHOW_ALL: `return all`,
  SHOW_MAIN: `show main page`,
  SHOW_DETAILS: `show film details page`,
  CHANGE_TAB: `change tab on film details page`,
  SHOW_MORE_FILMS: `show more films on main page`,
};

const ActionCreator = {
  // showAll: () => ({
  //   type: ActionType.SHOW_ALL,
  //   filter: Genres.All,
  // }),
  // filterByGenre: (filter) => ({
  //   type: ActionType.FILTER,
  //   filter,
  // }),
  showMain: () => ({
    type: ActionType.SHOW_MAIN,
    showedFilmId: null,
  }),
  showDetails: (id) => ({
    type: ActionType.SHOW_DETAILS,
    showedFilmId: id,
  }),
  changeTab: (tab) => ({
    type: ActionType.CHANGE_TAB,
    tab,
  }),
};

const initialState = {
  promoFilm,
  films: filmsData,
  // filter: Genres.ALL,
  screenMode: ScreenMode.MAIN,
  showedFilmId: null,
  currentTab: TabsType.OVERVIEW,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {

    case ActionType.SHOW_ALL:
      return extend(state, {films: initialState.films, filter: initialState.filter});

    case ActionType.FILTER:
      return extend(state, {
        films: initialState.films.filter((element) => {
          return element.genre === action.filter;
        }),
        filter: action.filter,
      });

    case ActionType.SHOW_DETAILS:
      return extend(state, {screenMode: ScreenMode.DETAILS, showedFilmId: action.showedFilmId});

    case ActionType.SHOW_MAIN:
      return extend(state, {screenMode: ScreenMode.MAIN, showedFilmId: action.showedFilmId});

    case ActionType.CHANGE_TAB:
      return extend(state, {currentTab: action.tab});

  }
  return state;
};

export {ActionType, ActionCreator, reducer};
