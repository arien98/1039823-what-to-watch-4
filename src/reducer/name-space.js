export const NameSpace = {
  DATA: `DATA`,
  SCREEN: `SCREEN`,
  USER: `USER`,
};
