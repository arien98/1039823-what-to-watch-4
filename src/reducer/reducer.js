import {combineReducers} from "redux";
import {reducer as data} from "./data/data.js";
import {reducer as screen} from "./screen/screen.js";
import {reducer as user} from "./user/user.js";
import {NameSpace} from "./name-space.js";


export default combineReducers({
  [NameSpace.DATA]: data,
  [NameSpace.SCREEN]: screen,
  [NameSpace.USER]: user,
});
