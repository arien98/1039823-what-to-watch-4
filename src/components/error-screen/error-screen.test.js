import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import ErrorScreen from "./error-screen";
import {Router} from "react-router-dom";
import {history} from "../../history";
import {NameSpace} from "../../reducer/name-space";

const mockStore = configureStore([]);


it(`Correctly ErrorScreen render`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      errorType: `400`,
    },
  });

  const props = {
    errorType: `400`,
    errorText: `Ох уж эти ошибки`,
  };

  const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <ErrorScreen {...props}/>
            </Router>
          </Provider>
      )
      .toJSON();
  expect(tree).toMatchSnapshot();
});
