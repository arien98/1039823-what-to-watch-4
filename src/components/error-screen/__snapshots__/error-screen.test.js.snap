// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly ErrorScreen render 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <div
      className="sign-in__message"
    >
      <p>
        Ошибка 400: Вы ввели неверные данные
      </p>
    </div>
  </div>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 
        2020
         What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
